src/index.js

```diff
@@ -44,15 +44,17 @@ const handler = (req, res) => {
     if (err) {
       // eslint-disable-next-line no-console
       console.error(err);
-      res.writeHead(err.status || err.statusCode || 500);
-      res.end(err.message);
+      res.statusCode = err.status || err.statusCode || 500;
+      res.setHeader('Content-Type', 'application/json');
+      res.end(JSON.stringify({ error: err.message }));
       return;
     }
     if (!res.finished) {
       if (!res.headersSent) {
-        res.writeHead(404);
+        res.statusCode = 404;
       }
-      res.end(`'${req.url}' not found`);
+      res.setHeader('Content-Type', 'application/json');
```

Comment:
This needs to be in the `if` block; if `res.headersSent` is true then we cannot safely do `res.setHeader`.